File: CPP/Min_Cost_Climbing_Stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        
        int dptable[cost.size()];
        
        dptable[0] = cost[0];
        dptable[1] = cost[1];
        
        for(int i=2;i<cost.size();i++){
            
            dptable[i] = cost[i] + min(dptable[i-1],dptable[i-2]);
        }
        
        return min(dptable[cost.size()-1],dptable[cost.size()-2]);
        
        
    }
};

File: Python/min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        c = len(cost)
        dp = [-1] * (c + 2)
        dp[c] = dp[c + 1] = 0
        for n in range(c - 1, -1, -1):
            dp[n] = cost[n] + min(dp[n + 1], dp[n + 2])
        return min(dp[0], dp[1])
